e2e/actions/common.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package actions
+
+import (
+	"os"
+	"testing"
+
+	"github.com/sylabs/singularity/e2e/internal/e2e"
+	"github.com/sylabs/singularity/internal/pkg/util/fs"
+)
+
+func workspaceDirsGenerator(t *testing.T, hostCanaryDir, hostHomeDir, hostWorkDir, hostCanaryFile string) {
```

Comment:
The linter started complaining about how many lines of code e2e/actions/actions.go and e2e/actions/oci.go had in common (fair point, linter!), so I refactored this function out and put it in this new file.

It's not "maximalist" refactoring – perhaps more could be refactored out – but as things stand now, there _are_ differences between actions.go and oci.go in some of the other nearly-identical portions of the relevant code. (For example, the difference in `checkHostFn()`, discussed in a separate comment in oci.go).